sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -130,6 +130,7 @@ abstract class Optimizer(catalogManager: CatalogManager)
     val operatorOptimizationBatch: Seq[Batch] = {
       Batch("Operator Optimization before Inferring Filters", fixedPoint,
         operatorOptimizationRuleSet: _*) ::
+      Batch("Infer window group limit", Once, InferWindowGroupLimit) ::
```

Comment:
I think this is an unrelated change and is actually a regression. We may infer filters and it's better to run window limit optimization as late as possible. @beliefer can we revert this change about moving the rule?

Reply:
Got it.